### ISSUE TITLE
request to replace dependency `connection` with `crypton-connection` or other fork

### ISSUE THREAD

**Issue by peterbecich** [ndmitchell/hoogle#418]
Hello

I see `hoogle` depends on `connection`. I've been told `connection` is permanently archived so will always require `tls` < 1.7 https://github.com/kazu-yamamoto/crypton-connection/issues/1

This makes `hoogle` difficult to build in Nix, because Nix Packages are at `tls` 1.8 and beyond:
```
$ nix search nixpkgs/haskell-updates#haskellPackages tls
.
.
* legacyPackages.x86_64-linux.haskellPackages.tls (1.8.0)
  TLS/SSL protocol native implementation (Server and Client)
```
I expect this will become a problem for Nix tools such as [`haskell-flake`](https://github.com/srid/haskell-flake) which bundle `hoogle`, HLS, and other things.

Can the dependency on `connection` be replaced with up-to-date https://hackage.haskell.org/package/crypton-connection or something else?

Thank you

**Comment by peterbecich** [comment]
I see this has already been done on `master`
https://github.com/ndmitchell/hoogle/blob/0be38ee5e078e31ef7eabeaba255aed12ce7055d/hoogle.cabal#L56
Latest release still uses `connection` https://hackage.haskell.org/package/hoogle
Can a fix version be pushed?

**Comment by ndmitchell** [comment]
Just released 5.0.18.4
